src/daft-core/src/array/ops/image.rs

```diff
@@ -104,6 +104,70 @@ impl AsImageObj for FixedShapeImageArray {
     }
 }
 
+pub fn image_array_from_img_buffers_iter<'a, I>(
```

Comment:
Could we possibly implement `image_array_from_img_buffers` with this function or remove it entirely? Just don't want to have two implementations for the same logic.